wcfsetup/install/files/lib/data/user/online/UsersOnlineList.class.php

```diff
@@ -156,20 +157,23 @@ public function checkRecord() {
 	 * @return	boolean
 	 */
 	public static function isVisible($userID, $canViewOnlineStatus) {
-		if (WCF::getSession()->getPermission('admin.user.canViewInvisible') || $userID == WCF::getUser()->userID) return true;
```

Comment:
These two conditions should always cause the method to return true, regardless of any program logic or restrictions. Please restore the original line and have them return early just as before. The code below should be altered to no not check for these two conditions again.